File: JS/LeftAndRightSumDifferences.test.js
/**
 * @version JAVASCRIPT ECMAScript 6 
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
import TEST from 'tape'
import { leftRightDifference } from './LeftAndRightSumDifferences.js'

TEST('Starting LeftAndRightSumDifferences test...', (t) => {
    //Test 1
    let want = [15,1,11,22]
    let got = leftRightDifference( [10,4,8,3] )
    t.assert( want.toString() === got.toString(), "Expect: "+want)
    //Test 2
    want = [0]
    got = leftRightDifference( [1] )
    t.assert( want.toString() === got.toString(), "Expect: "+want)

    t.end()
})
  

File: JAVA/LeftAndRightSumDifferencesTest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class LeftAndRightSumDifferencesTest {

    @Test
	public void testLeftRightDifference() throws Exception {
        //Test 1
        int[] want = {15,1,11,22};
        int[] got = LeftAndRightSumDifferences.leftRightDifference( new int[]{10,4,8,3} );
        assertEquals(java.util.Arrays.toString(want), java.util.Arrays.toString(got));
        //Test 2
        want = new int[] {0};
        got = LeftAndRightSumDifferences.leftRightDifference( new int[]{1} );
        assertEquals(java.util.Arrays.toString(want), java.util.Arrays.toString(got));
	}

}
